lib/core/widgets/selection_button.dart

```diff
@@ -12,12 +9,14 @@ class SelectionButton extends StatefulWidget {
   final bool selected;
   final LinearGradient gradient;
   final TextStyle textStyle;
+  final Widget child;
```

Comment:
In deinem Code sehe ich, dass du entweder das child anzeigst oder das Label. Am besten wäre es, wenn man als Dev das auch anhand der Codedokumentation erkennen kann, also wenn man in der IDE über den Parameter des Konstruktors hovered. 

Hier am besten noch ein kurzer Dokumentationskommentar eine Zeile drüber (mit /// vorne und auf **englisch**), damit andere Entwickler besser verstehen, was das child für Auswirkungen hat. 

- [ ] Einfach ein kurzer Satz, dass das das Child ist, das auf dem Button angezeigt wird und dass wenn dieses child nicht null ist, das Label nicht angezeigt wird.

[Mehr Infos](https://dart.dev/guides/language/effective-dart/documentation#do-use--doc-comments-to-document-members-and-types)